Update menu_lua_api.txt to replace deprecated functions

ClobberXD opened the pull request:
This minor PR replaces all occurrences of `core.setting_*` (deprecated) with `core.settings:*`
in `doc/menu_lua_api.txt`

SmallJoker commented on doc/menu_lua_api.txt:
```diff
@@ -186,11 +186,11 @@ core.log(loglevel, line) (possible in async calls)
+core.settings:set(name, value)
+core.settings:get(name) -> string or nil (possible in async calls)
+core.settings:setbool(name, value)
+core.settings:getbool(name) -> bool or nil (possible in async calls)
```
`get_bool` and `set_bool`
It's also incomplete and easily forgotten in changes. Perhaps we should simply refer to the `lua_api.txt` docs for the settings object.

ClobberXD commented:
@SmallJoker even lua_api.txt does not list out all the member functions of the `minetest.settings` object. I'll make the appropriate changes to lua_api.txt too, if that's ok with you.

Currently, this is all we have:
https://github.com/minetest/minetest/blob/master/doc/lua_api.txt#L2973

SmallJoker commented:
> Even lua_api.txt does not list out all the member functions of the minetest.settings object. 

[Sorry?](https://github.com/minetest/minetest/blob/master/doc/lua_api.txt#L4759-L4784)

ClobberXD commented:
My bad, sorry.

Added a link to that section in `menu_lua_api.txt`, and removed dummy entry at the bottom of the file.
Does the wording/format require any changes for consistency with the rest of the file?

NewbProgrammer101 commented on doc/menu_lua_api.txt:
```diff
@@ -192,6 +192,9 @@ core.settings:set_bool(name, value)
 core.settings:save() -> nil, save all settings to config file
 
+For a more comprehensive list of methods of the Settings object see
+[lua_api.txt](https://github.com/minetest/minetest/blob/master/doc/lua_api.txt#L4759)
```
That's only for Markdown. This is in plain text.

Just drop the title and parentheses. Leave the link as is.

SmallJoker replied:
`lua_api.txt` and `client_lua_api.txt` are both also text files, but written so they can be read readily with a markdown application. This file was not changed to that format but markdown links seem acceptable here too.

ClobberXD replied:
I agree with @SmallJoker here. All the `*lua_api.txt` files are in markdown format, but have the `.txt` extension for easy linking to specific lines. e.g. https://github.com/minetest/minetest/blob/master/doc/lua_api.txt#L128

paramat replied:
Yes MD format is fine here.

NewbProgrammer101 replied:
Ok, got it. Sorry for wasting your time!

NewbProgrammer101 commented:
Also, rebase please.

`git pull origin master`

paramat commented on doc/menu_lua_api.txt:
```diff
@@ -1,5 +1,5 @@
 Minetest Lua Mainmenu API Reference 0.5.0
```
5.0.0

SmallJoker replied:
Will be solved by rebasing. This branch is simply outdated.

ClobberXD replied:
Done, thanks for pointing that out.

paramat commented:
The PR now seems to contain our version change commit?

ClobberXD commented:
I'm very sorry. I've messed up the rebasing part... I'll just make a new PR for this.
